flink-runtime/src/main/java/org/apache/flink/runtime/memory/MemoryManager.java

```diff
@@ -568,144 +569,26 @@ public int computeNumberOfPages(double fraction) {
 			throw new IllegalArgumentException("The fraction of memory to allocate must within (0, 1].");
 		}
 
+		//noinspection NumericCastThatLosesPrecision
 		return (int) (totalNumPages * fraction / numberOfSlots);
 	}
 
```

Comment:
I think we also need 'computeNumberOfPages' for specific memory type, for the same reason as 'allocatePages'.

Reply:
I still think that the fraction concern is not strictly related to the memory management per se but it is rather a higher level concern about how the memory is distributed among the operators. I think the main reason we had this method was even to hold the number of slots which should go away when the memory manager is per slot.

Fractions and the memory limits can be handled in one place where the operators get the memory manager and their limits which they can decide how to break into chunks and pages.
Anyways, I agree we can reiterate on this later during the MM per slot refactoring.

Instead, I suggest we expose the total number of pages per type at the moment, same as memory size per type which is naturally related to the memory management.

Reply:
I think the reason we had this method was not only hold the number of slots, but also hold the total number of pages. In other words, MM knows how many memory in total the slot has, which will not change when we have MM per slot.

I agree that computing operator level memory budget from the slot total memory and the operator fraction is another responsibility for MM. It makes sense that MM exposes the slot total memory and operator calls a util to compute the budget from the slot total memory and the fraction.